src/mmap.rs

```diff
@@ -125,33 +138,34 @@ pub fn check_file_offset(
 /// Represents a continuous region of the guest's physical memory that is backed by a mapping
 /// in the virtual address space of the calling process.
 #[derive(Debug)]
-pub struct GuestRegionMmap {
-    mapping: MmapRegion,
+pub struct GuestRegionMmap<B> {
```

Comment:
Use default generic type parameter may reduce the changes needed.
```
pub struct GuestRegionMmap<B = ()> {
}
```
Also suggest to change to
```
pub struct GuestRegionMmap<T = GuestMemoryAssociatedTypes> {
}
```
to in case we need to add more associated types in future.